From c73a95cb7ab29228ebb2099ed30067d2bba8294f Mon Sep 17 00:00:00 2001
From: Bruce Forstall <brucefo@microsoft.com>
Date: Fri, 8 Dec 2017 18:07:05 -0800
Subject: [PATCH] Do not optimize flags-setting add

In minopts, we were attempting to generate "add r0,r0,0" and
optimizing it away completely. But it if needs to generate
the flags, for a subsequent overflow (flags) check, we can't do that.

Fixes #14860
---
 src/jit/emitarm.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/jit/emitarm.cpp b/src/jit/emitarm.cpp
index a0a2ee7..a35c729 100644
--- a/src/jit/emitarm.cpp
+++ b/src/jit/emitarm.cpp
@@ -2392,7 +2392,7 @@ void emitter::emitIns_R_R_I(instruction ins,
             assert(insOptsNone(opt));
 
             // Is it just a mov?
-            if (imm == 0)
+            if ((imm == 0) && insDoesNotSetFlags(flags))
             {
                 // Is the mov even necessary?
                 // Fix 383915 ARM ILGEN
-- 
2.7.4

